go-generate and go-test need to be recreated if any of their template inputs change

### Version of Apptainer:
What version of Apptainer are you using? Run:
```
$ apptainer --version
apptainer version 0.1.1+125-g0b671cb16-dirty
```

### Expected behavior
Commands:
```
./mconfig
cd buiddir
make
```
Expected to run go as found in previous step.

### Actual behavior
First command finds go correctly. The make command expects go to be located in `/usr/bin/go`

<!-- if this is a feature request, you can ignore this next part -->

### Steps to reproduce this behavior
Install go following instructions here: https://go.dev/doc/install
Install apptainer:
```
./mconfig
cd buiddir
make
```

### What OS/distro are you running
```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

```

### How did you install Apptainer
source

Yes I re-ran mconfig. Here are steps that should be reproducible. Assuming a fresh apptainer repository and no go installed.
```bash
wget https://go.dev/dl/go1.17.6.linux-amd64.tar.gz
sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.17.6.linux-amd64.tar.gz
echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc
source ~/.bashrc

cd path/to/apptainer
./mconfig
cd builddir
make
sudo make install

sudo mv /usr/local/go /usr/local/go2
echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc
source ~/.bashrc
cd path/to/apptainer
./mconfig
cd builddir
make
```

The last command errors and will output:
```
 GEN GO DEP /home/davis/Documents/Programs/apptainer/builddir/starter.d
 GEN GO DEP /home/davis/Documents/Programs/apptainer/builddir/singularity.d
/home/davis/Documents/Programs/apptainer/scripts/go-generate: 9: exec: /usr/local/go/bin/go: not found
make: *** [Makefile:160: /home/davis/Documents/Programs/apptainer/internal/pkg/buildcfg/config.go] Error 127
```

The install instructions are found here: https://go.dev/doc/install

--- a/mlocal/frags/build_scripts.mk
+++ b/mlocal/frags/build_scripts.mk
@@ -6,7 +6,7 @@ $(SOURCEDIR)/scripts/go-test: export GO111MODULE := $(GO111MODULE)
 $(SOURCEDIR)/scripts/go-test: export GOFLAGS := $(GOFLAGS)
 $(SOURCEDIR)/scripts/go-test: export GO_TAGS := $(GO_TAGS)
 $(SOURCEDIR)/scripts/go-test: export SUDO_SCRIPT := $(SOURCEDIR)/scripts/test-sudo
-$(SOURCEDIR)/scripts/go-test: $(SOURCEDIR)/scripts/go-test.in $(SOURCEDIR)/scripts/expand-env.go
+$(SOURCEDIR)/scripts/go-test: $(SOURCEDIR)/scripts/go-test.in $(SOURCEDIR)/scripts/expand-env.go $(BUILDDIR)/config.h
 	@echo ' GEN $@'
 	$(V) $(GO) run $(GO_MODFLAGS) $(SOURCEDIR)/scripts/expand-env.go < $< > $@
 	$(V) chmod +x $@
@@ -19,7 +19,7 @@ $(SOURCEDIR)/scripts/go-generate: export GO := $(GO)
 $(SOURCEDIR)/scripts/go-generate: export GO111MODULE := $(GO111MODULE)
 $(SOURCEDIR)/scripts/go-generate: export GOFLAGS := $(GOFLAGS)
 $(SOURCEDIR)/scripts/go-generate: export GO_TAGS := $(GO_TAGS)
-$(SOURCEDIR)/scripts/go-generate: $(SOURCEDIR)/scripts/go-generate.in $(SOURCEDIR)/scripts/expand-env.go
+$(SOURCEDIR)/scripts/go-generate: $(SOURCEDIR)/scripts/go-generate.in $(SOURCEDIR)/scripts/expand-env.go $(BUILDDIR)/config.h
 	@echo ' GEN $@'
 	$(V) $(GO) run $(GO_MODFLAGS) $(SOURCEDIR)/scripts/expand-env.go < $< > $@
 	$(V) chmod +x $@

